### DISCUSSION TITLE
Webpack fails when importing module

### DISCUSSION THREAD

**Discussion by ninjamar** [webpack/webpack#18335] [Q&A] [answered]
# Bug report

**What is the current behavior?**
I try to import the dedent module
```javascript
import dedent from 'https://cdn.jsdelivr.net/npm/dedent@1.5.3/+esm';
```
but webpack fails with
```
ERROR in external "https://cdn.jsdelivr.net/npm/dedent@1.5.3/+esm"
The target environment doesn't support dynamic import() syntax so it's not possible to use external type 'module' within a script
Error: The target environment doesn't support dynamic import() syntax so it's not possible to use external type 'module' within a script
```
webpack.config.js
```javascript
module.export = {
    entry: "./foo.js"
}
```
foo.js
```javascript
import dedent from 'https://cdn.jsdelivr.net/npm/dedent@1.5.3/+esm';

dedent`
    Foo
    Bar
    Baz
`;
```
**If the current behavior is a bug, please provide the steps to reproduce.**
Run webpack on code that has `import dedent from 'https://cdn.jsdelivr.net/npm/dedent@1.5.3/+esm';`

**What is the expected behavior?**
The expected behavior is that webpack bundles the module.

**Other relevant information:**
webpack version:  5.91.0
Node.js version: 17.9.1
Operating System: macOS 14.1.1 (M1)
Additional tools:

**Comment by alexander-akait** [comment]
Beucase by default we don't know your environment (I don't know about `import(...)` support), so you need to set `target` or enable ES modules output

**Reply by alexander-akait** [reply]
Here you can find how we undestand it - https://github.com/webpack/webpack/blob/main/lib/config/target.js#L311

**Reply by alexander-akait** [reply, marked as answer]
So if you need to bundle for browser, you need to set `target: ["web", "es2020"]` or you can use `browserslist` and we automatically load it

**Reply by ninjamar** [reply]
Thank you

**Reply by mbrodala** [reply]
Can you explain in more detail what should be in `browserslist` specifically to enable ES2020 and thus dynamic import?
